tools/disass/ida/disass.py

```diff
@@ -136,6 +139,87 @@ def get_direct_branch_target(branch_inst_ea):
         branch_inst_ea, target_ea))
     return target_ea
 
+# source: http://reverseengineering.stackexchange.com/a/1648/4197
```

Comment:
Regarding program.get_basic_block, I never was able to get it to return a basic block that had data. has_basic_block would return False and any call to get_basic_block, with an arbitrary address, or the address of the first instruction of a basic block, would return a basic block with a bunch of null fields. I'll reproduce this on Monday and open a bug.

On indirect calls I could not find a way to check if the target function was noreturn. The flags were always -1. Is there a way to dereference the target address, check if it is in .idata, then get flags from the target in there? I'll give that a shot to see if I can get flags. The other solution would be getting the name of the target function and keeping a black list of noreturn names, that doesn't sound ideal though. I reverted to structural verification because of this, and I reverted to IDA's flow chart due to the get_basic_block not returning initialized blocks.

Could each instruction store a reference to its parent block? That would make retrieval most simple. Rather than going from arbitrary address, then using IDA to get the first address in the basic block, then retrieving the basic block from get_basic_block.